qiskit_experiments/framework/base_experiment.py

```diff
@@ -26,26 +27,18 @@
 from qiskit.qobj.utils import MeasLevel
 from qiskit.providers.options import Options
 from qiskit_experiments.framework.store_init_args import StoreInitArgs
+from qiskit_experiments.framework.base_analysis import BaseAnalysis
 from qiskit_experiments.framework.experiment_data import ExperimentData
 from qiskit_experiments.framework.configs import ExperimentConfig
 
 
 class BaseExperiment(ABC, StoreInitArgs):
-    """Abstract base class for experiments.
-
-    Class Attributes:
-
-        __analysis_class__: Optional, the default Analysis class to use for
-                            data analysis. If None no data analysis will be
-                            done on experiment data (Default: None).
-    """
-
-    # Analysis class for experiment
-    __analysis_class__ = None
+    """Abstract base class for experiments."""
 
     def __init__(
         self,
         qubits: Sequence[int],
+        analysis: Optional[BaseAnalysis] = None,
```

Comment:
I like it! This will increase flexibility of workflow. However, can we directly set callback to experiment data, i.e. not limited to analysis? For example, 

```python
def __init__(self, qubits, backend, experiment_type, callbacks: List[Callable]):
```
if we run two different analysis, we can write
```python
MyExperiment([0, 1], my_backend, [MyAnalysis1(), MyAnalysis2()])
```
or we can do whatever we want
```python
MyExperiment([0, 1], my_backend, [MyAnalysis(), my_custom_validate_analysis()])
```
probably this will be a real headache for serialization.

Reply:
Lets save this for later discussion since it sounds quite complicated. I would argue that you should do that sort of thing by defining a custom analysis class that can combine other analysis classes to keep experiment class straight forward.